### DISCUSSION TITLE
Searching functionality in developer portal

### DISCUSSION THREAD

**Discussion by Nazarah** [Azure/api-management-developer-portal#2560] [Feature request]
Hi,

I am trying out the CMS capabilities of Azure APIM Developer Portal.
I understand that for page types `Products` and `APIs`, there is a search functionality to look for relevant APIs or Products added as part of the APIM.
Is there any searching functionality to look for content added as part of  the CMS?

e.g. I am trying to design a listing page for developer guides and I want to add a keyword-based searching option to find relevant guides.
Is it possible to add such an option in a listing page apart from Product or API?

<img width="1679" alt="Screen Shot 2020-11-02 at 21 30 52" src="https://user-images.githubusercontent.com/17976243/97910700-c6841200-1d52-11eb-87ba-ee290f3f3f9d.png">

**Comment by mikebudzynski** [comment]
@Nazarah There's already a feature request to implement static content search (#881), which I just merged into this issue. At this moment, the developer portal doesn't support this functionality and we don't have an ETA.

**Comment by mikebudzynski** [comment]
From @AnRei123 in #881:

> Yes, the missing search functionality for static content is also an issue for and urgently needed as our amount of static content is increasing continuously. 
> 
> 1) The search should allow to filter for all page titles that contain a dedicated string. Matches in the title shall be ranked higher than matches in the description. This must be reflected in the list with the search results.
> 2) The search should allow to filter for all page contents that a dedicated string or a combination of several strings. (searching for one string might not always sufficient. 
> The search result should list the page titles and offer an extension to see the first three sentences of a page. Navigating to the search target should highlight the the strings on the page to easier navigate on the target page.
> If no string is entered in the search field, the complete list with all static pages should be listed in a scrollable manner to allow getting an overview and an impression on the available and amounts of contents. The titles of the pages should be listed in an alphabetically order.
> 3) A configurable web widget is required for add a search field in the header that also contains the main navigation.
> 4) The search shall also allow to search not only in the static content, but also in the descriptions of API products, APIs and operations according to the access rights of the DevPortal user. As we have a mixed approach to document our APIs, it is very important that all contents of the DevPortal are considered for the full-text search - the static page content and the dynamic API content including names of operations and all kinds of descriptions.  The search widget shall be configurable accordingly. There should be also offered different sizes and appearances for the different placements on the WebPortal page. 
> The search widget should be configurable, if the search should search for dynamic API contents only or static content or a combination of it. The size (length, width) of the search field and the display of the zoom icon should be configurable as well.

**Comment by frankdewaelle** [comment]
Hi, is this feature request still on any planning?

**Comment by mikebudzynski** [comment]
@frankdewaelle, this feature is on our radar but it's not on the immediate roadmap for the next few months.

**Comment by frankdewaelle** [comment]
@mikebudzynski Thanks for the feedback. Any way we notice such an update (version or notification) ?

**Comment by mikebudzynski** [comment]
The best way to stay updated is to subscribe to this issue for notifications. We also post detailed release notes in the [Releases section of this repository](https://aka.ms/apimdevportal/releases) as well as in [the main API Management releases repository](https://aka.ms/apim/releases).

**Comment by Mdshakeelse** [comment]
Hi Team, any update on the search functionality for new developer portal, I think this must needed functionality I believe. We are currently blocked to get migrated to new developer portal, let us know if any tentative dates or any work around. Thanks
